### PR TITLE
[Unity][Frontend][NN] Conv2D NN module

### PR DESCRIPTION
**Pull request by jwfromm**
In this PR I add support for `Conv2D` in our torch-like nn api. I'm planning on adding quite a few other such layers but wanted to get feedback on this one to make sure there arent any mistakes that would propagate :)

### CODE REVIEW

**Review by junrushao — APPROVED** [review]
Neat. Thanks!

**Review by cyx-6 — APPROVED** [review]
It looks great! Just keep uploading other layers in need. :)

File: python/tvm/relax/frontend/nn/modules.py [hunk 1]
```diff
@@ -107,6 +107,65 @@ def forward(self, x: Tensor) -> Tensor:  # pylint: disable=invalid-name
...
+
+        Parameters
+        ----------
+        x : Tensor
+            The input tensor.
+
+        Returns
+        -------
+        ret : Tensor
+            The output tensor for the linear layer.
```
Comment (LeshengJin) [inline python/tvm/relax/frontend/nn/modules.py:162]: A small typo

**Review by LeshengJin — APPROVED** [review]
LGTM

**Pull request merged**